app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -78,11 +79,18 @@ def perform_reboot_all hosts = Host::Discovered.all
             error =  "#{discovered_host.name}: failed to reboot "
             overall_errors << error
             logger.error error
+            host_details << {:name => "#{discovered_host.name}", :error => "#{error}"}
           end
         rescue Exception => e
-          error = "#{discovered_host.name}: #{e.to_s} "
-          overall_errors << error
+          error = "#{e.to_s} "
+          msg = if e.respond_to?(:wrapped_exception)
+                  e.wrapped_exception.message
+                else
+                  e.message
+                end
```

Comment:
This does not appear to be correct, you convert Exception class to String and then for `wrapped_exception` method presence. Lemme explain, in Foreman we have Foreman::Exception and WrappedException defined. These are just exception wrappers with error codes (ERF-XXX-XXX) which allows us to maintain list of codes. Now, WrappedException is used when you want to store the original exception inside, that's why it's useful.

https://github.com/theforeman/foreman/blob/develop/lib/foreman/exception.rb

So if you catch a generic exception, it will either contain message or wrapped_exception.message does it make any sense? So you want to call respond_to? on `e` directly rather than converting it to string.

https://projects.theforeman.org/projects/foreman/wiki/ErrorCodes